### PR TITLE
Fixing animation sampling

### PR DESCRIPTION
**Pull request by mattgperry**
Fixes https://github.com/framer/motion/issues/1867

### CODE REVIEW

File: packages/framer-motion/src/animation/waapi/create-accelerated-animation.ts [hunk 1]
```diff
@@ -88,10 +89,10 @@ export function createAcceleratedAnimation(
          */
         const { currentTime } = animation
         if (currentTime) {
-            const sampleAnimation = animate(options)
+            const sampleAnimation = animate({ ...options, autoplay: false })
```
Comment (shuangq) [inline packages/framer-motion/src/animation/waapi/create-accelerated-animation.ts:92]: What does `autoplay: false` do here?

File: packages/framer-motion/src/animation/legacy-popmotion/index.ts [hunk 2]
```diff
@@ -164,10 +165,27 @@ export function animate<V = number>({
...
+         * temporal-resolution version.
+         */
+        sample: (t: number): V => {
+            elapsed = initialElapsed
+            const sampleResolution =
+                typeof duration === "number" ? duration * 0.5 : 50
+
+            let sampleElapsed = 0
+            update(0)
+            while (sampleElapsed <= t) {
```
[the remark this answers is not in the document]
Comment (mattgperry) [reply packages/framer-motion/src/animation/legacy-popmotion/index.ts:182]: Ah yeah this is a great point, let me add a test for this.

File: packages/framer-motion/jest.setup.tsx [hunk 3]
```diff
@@ -30,6 +30,10 @@ if (!(global as any).ResizeObserver) {
     } as any
 }
 
+// Stub scrollTo
+const noop = () => {}
+Object.defineProperty(global, "scrollTo", { value: noop, writable: true })
+
```
Comment (shuangq) [inline packages/framer-motion/jest.setup.tsx:36]: Why do we need this in the PR?

**Review by shuangq — APPROVED** [review]

File: packages/framer-motion/jest.setup.tsx (hunk 3 above)
Comment (mattgperry) [reply packages/framer-motion/jest.setup.tsx:36]: It's unrelated but removes some noise from the Jest console output

**Pull request merged**